File: springboot-04-springapplication/src/main/java/com/linkedbear/springboot/SpringApplicationApplication.java
package com.linkedbear.springboot;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class SpringApplicationApplication {

    public static void main(String[] args) {
        SpringApplication.run(SpringApplicationApplication.class, args);

//        SpringApplication springApplication = new SpringApplication();
//        springApplication.setMainApplicationClass(SpringApplicationApplication.class);
//        springApplication.setBannerMode(Banner.Mode.OFF);
//        springApplication.run(args);
//
//        new SpringApplicationBuilder(SpringApplicationApplication.class)
//                .bannerMode(Banner.Mode.OFF)
//                .web(WebApplicationType.NONE)
//                .run(args);

//        new SpringApplicationBuilder(SpringApplicationApplication.class).listeners(new ApplicationStartingListener()).run(args);
    }
}

File: springboot-04-springapplication/src/main/kotlin/com/david/springboot/SpringApplicationApplication.kt
package com.david.springboot

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication

@SpringBootApplication
open class SpringApplicationApplication

fun main(args: Array<String>) {
    SpringApplication.run(SpringApplicationApplication::class.java, *args)

    //  直接操作 API
//      val springApplication = SpringApplication()
//      springApplication.sources = setOf(SpringApplicationApplication::class.java.name)
//      springApplication.setBannerMode(Banner.Mode.OFF)
//      springApplication.run(*args)

    //  借助构造器
//    SpringApplicationBuilder(SpringApplicationApplication::class.java)
//        .bannerMode(Banner.Mode.OFF)
//        .web(WebApplicationType.NONE)
//        .run(*args)

}